### PR TITLE
Allow empty argument to GetKeyBindings instead

### PR DESCRIPTION
**Pull request by badosu**
Keep the API concise, the prefix matching is not essential for perf for
the use cases it would use anyways. Keybinding check would be done at
sparse callins or initialization

### CODE REVIEW

File: rts/Game/UI/KeyBindings.cpp [hunk 1]
```diff
@@ -305,19 +305,15 @@ void CKeyBindings::Kill()
 
 
 /******************************************************************************/
-const CKeyBindings::ActionList& CKeyBindings::GetActionList(const std::string& matcher) const
+const CKeyBindings::ActionList& CKeyBindings::GetActionList() const
 {
 	static ActionList merged; //FIXME switch to thread_local (?)
```
Comment (badosu) [inline rts/Game/UI/KeyBindings.cpp:310]: This problem can be consolidated into one declaration instead, I copied the previous method. Notice also the ported FIXME, which I'm not sure what it refers to.

File: rts/Lua/LuaUnsyncedRead.cpp [hunk 2]
```diff
@@ -2239,33 +2238,21 @@ int LuaUnsyncedRead::GetKeySymbol(lua_State* L)
...
-		lua_rawseti(L, -2, i++);
-	}
-	return 1;
-}
-
 int LuaUnsyncedRead::GetKeyBindings(lua_State* L)
 {
-	CKeySet ks;
+	CKeyBindings::ActionList actions;
+	const std::string& argument = luaL_optstring(L, 1, "");
```
Comment (badosu) [inline rts/Lua/LuaUnsyncedRead.cpp:2244]: This is basically the same as previous but just handling optional args. The additional read was removed.

File: rts/Game/UI/KeyBindings.cpp [hunk 3]
```diff
@@ -305,19 +305,15 @@ void CKeyBindings::Kill()
...
 	const ActionList* alPtr;
 
 	for (const auto& p: bindings) {
```
Comment (badosu) [inline rts/Game/UI/KeyBindings.cpp:313]: There could probably be a map of a map to be more efficient, I'm unfamiliar with boost so this is a simple implementation.

File: rts/Game/UI/KeyBindings.cpp (hunk 1 above)
Comment (marcushutchings) [reply rts/Game/UI/KeyBindings.cpp:310]: I don't think the FIXME is a concern at the moment. It isn't clear how thread local storage would be helpful at the moment.

**Review by marcushutchings — APPROVED** [review]
Simple enough changes; though, I notice I've caught the second PR of two, which confused me a little.

**Pull request merged**